src/terminal.rs

```diff
@@ -22,26 +22,18 @@ pub trait Terminal {
         let _ = terminal::disable_raw_mode()?;
         Ok(())
     }
-}
 
-/// TerminalEvent is a wrapper of crossterm::event.
-/// This is intended for mocking terminal-specific functions.
-pub trait TerminalEvent {
     fn poll(&self, timeout: Duration) -> Result<bool> {
         let b = event::poll(timeout)?;
         Ok(b)
     }
 
-    fn read(&mut self) -> Result<Event> {
+    fn read(&self) -> Result<Event> {
```

Comment:
I think this function doesn't have to be called in a context of a mutable reference.